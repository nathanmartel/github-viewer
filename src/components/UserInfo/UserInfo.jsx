import React from 'react';
import PropTypes from 'prop-types';

const UserInfo = ({ name, avatar_url, followers, following, url }) => (
  <>
    { avatar_url &&
    <div className='user-info-photo'>
      <img src={avatar_url} alt={name} />
    </div> }
    { name && 
    <div className='user-info-profile'>
      <h2>{name}</h2>
      { followers && <p>Followers: {followers}</p> }
      { following && <p>Followers: {following}</p> }
      { url && <p><a href={url}>GitHub profile</a></p> }
    </div> }
  </>
);

UserInfo.propTypes = {
  name: PropTypes.string.isRequired,
  avatar_url: PropTypes.string.isRequired,
  followers: PropTypes.number.isRequired,
  following: PropTypes.number.isRequired,
  url: PropTypes.string.isRequired,
};

export default UserInfo;
