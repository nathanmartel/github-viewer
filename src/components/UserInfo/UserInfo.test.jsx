import React from 'react';
import { shallow } from 'enzyme';
import UserInfo from './UserInfo';

describe('UserInfo component', () => {
  it('renders UserInfo', () => {
    const userInfo = { 
      name: 'John Doe',
      avatar_url: 'https://github.com/images/error/octocat_happy.gif',
      followers: 1,
      following: 12,
      url: 'http://www.hello.com'
    };
    const wrapper = shallow(<UserInfo {...userInfo} />);
    expect(wrapper).toMatchSnapshot();
  });
});
