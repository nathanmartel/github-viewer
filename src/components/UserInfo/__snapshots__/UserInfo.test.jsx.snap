// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UserInfo component renders UserInfo 1`] = `
<Fragment>
  <div
    className="user-info-photo"
  >
    <img
      alt="John Doe"
      src="https://placekitten.com/200/200"
    />
  </div>
  <div
    className="user-info-profile"
  >
    <h2>
      John Doe
    </h2>
    <p>
      Followers: 
      1
       | Following: 
      12
    </p>
    <p>
      <a
        href="http://www.hello.com"
      >
        GitHub profile
      </a>
    </p>
  </div>
</Fragment>
`;
